bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -84,18 +87,20 @@ public BlueZBluetoothDevice(BlueZBridgeHandler adapter, tinyb.BluetoothDevice ti
      *
      * This method should always be called directly after creating a new object instance.
      */
-    public synchronized void initialize() {
-        scheduler.submit(() -> {
-            if (this.device == null) {
-                tinyb.BluetoothDevice tinybDevice = findTinybDevice(address.toString());
-                if (tinybDevice != null) {
-                    device = tinybDevice;
-                    enableNotifications();
-                }
-            } else {
+    public void initialize() {
+        scheduler.submit(this::initializeDevice);
+    }
+
+    private synchronized void initializeDevice() {
+        if (this.device == null) {
+            tinyb.BluetoothDevice tinybDevice = findTinybDevice(address.toString());
+            if (tinybDevice != null) {
+                device = tinybDevice;
                 enableNotifications();
             }
-        });
+        } else {
+            enableNotifications();
```

Comment:
as both branches end with `enableNotifications`, shouldn't this be moved out of the `if` completely?

Reply:
`enableNotifications` can't be called in the case that the `device` is null otherwise you get a NPE. The inner if statement of the above branch checks for that so there is a chance that `enableNotifications` would never get called if a tinyb.BluetoothDevice could not be found.
But more importantly, I didn't write this logic, I was just moving it to its own method so it could be synchronized properly.